articles/quickstart/backend/rails/01-authorization.md

```diff
@@ -30,154 +30,256 @@ gem 'jwt'
+Create a Concern called `Secured` which looks for the Access Token in the `Authorization` header of an incoming request. If the token is present, it should be passed to `Auth0Client.validate_token`.
 
 ```rb
 # app/controllers/concerns/secured.rb
 
 # frozen_string_literal: true
+
 module Secured
   extend ActiveSupport::Concern
 
-  included do
-    before_action :authenticate_request!
+  REQUIRES_AUTHENTICATION = { message: 'Requires authentication' }.freeze
+  BAD_CREDENTIALS = {
+    message: 'Bad credentials'
+  }.freeze
+  MALFORMED_AUTHORIZATION_HEADER = {
+    error: 'invalid_request',
+    error_description: 'Authorization header value must follow this format: Bearer access-token',
+    message: 'Bad credentials'
+  }.freeze
+
+  def authorize
+    token = token_from_request
+
+    return if performed?
+    
+    validation_response = Auth0Client.validate_token(token)
+    
+    return unless (error = validation_response.error)
+    
+    render json: {message: error.message}, status: error.status
   end
 
   private
 
-  def authenticate_request!
-    auth_token
-  rescue JWT::VerificationError, JWT::DecodeError
-    render json: { errors: ['Not Authenticated'] }, status: :unauthorized
-  end
+  def token_from_request
+    authorization_header_elements = request.headers['Authorization']&.split
 
-  def http_token
-    if request.headers['Authorization'].present?
-      request.headers['Authorization'].split(' ').last
+    render json: REQUIRES_AUTHENTICATION, status: :unauthorized and return unless authorization_header_elements
+
+    unless authorization_header_elements.length == 2
+      render json: MALFORMED_AUTHORIZATION_HEADER, status: :unauthorized and return
     end
-  end
 
-  def auth_token
-    JsonWebToken.verify(http_token)
+    scheme, token = authorization_header_elements
+
+    render json: BAD_CREDENTIALS, status: :unauthorized and return unless scheme.downcase == 'bearer'
+
+    token
   end
 end
 ```
 
 ### Validate scopes
 
-The `JsonWebToken.verify` method above verifies that the Access Token included in the request is valid; however, it doesn't yet include any mechanism for checking that the token has the sufficient **scope** to access the requested resources.
+The `Auth0Client.validate_token` method above verifies that the Access Token included in the request is valid; however, it doesn't yet include any mechanism for checking that the token has the sufficient **scope** to access the requested resources.
 
-To look for a particular `scope` in an Access Token, provide an array of required scopes and check if they are present in the payload of the token.
+To look for a particular `scope` in an Access Token, create a new struct in your `Auth0Client` class called `Token` and define a new method inside, `validate_permissions`, that given an array of required scopes it will check if they are present in the payload of the token.
```

Comment:
```suggestion
To look for a particular `scope` in an access token, create a new struct in your `Auth0Client` class called `Token` and define a new method inside, `validate_permissions`, that given an array of required scopes it will check if they are present in the payload of the token.
```
Same as above.